### PR TITLE
[module] find: Add ability to exclude results by pattern.

### PR DESCRIPTION
**Pull request by k0ste**
#### SUMMARY

Add ability to exclude results by pattern. 

##### ISSUE TYPE

Feature Pull Request

##### COMPONENT NAME

modules/find

##### ANSIBLE VERSION

```
ansible 2.2.2.0
  config file = /home/k0ste/.ansible.cfg
  configured module search path = ['/home/k0ste/ansible/my_modules/']
```

##### ADDITIONAL INFORMATION

Without regex:
``` yaml
---
- name: jedi
  become: false
  gather_facts: fasle
  remote_user: k0ste
  no_log: false
  strategy: free
  hosts: localhost
  tasks:
  - name: "Step 1"
    find:
      paths: "/var/log"
      recurse: "no"
      file_type: "file"
      patterns: "*.log"
      excludes: "pacman.log"
    register: "find_result"
  - name: "Step 2"
    debug:
      var: "vars['find_result']['files']"
```

Without excludes:
```
[k0ste@WorkStation tmp]$ ansible-playbook jedi.yml | grep path
            "path": "/var/log/minidlna.log", 
            "path": "/var/log/pacman.log", 
            "path": "/var/log/lxdm.log", 
            "path": "/var/log/Xorg.0.log", 
```
With excludes:
```
[k0ste@WorkStation tmp]$ ansible-playbook jedi.yml | grep path
            "path": "/var/log/minidlna.log", 
            "path": "/var/log/lxdm.log", 
            "path": "/var/log/Xorg.0.log", 
```
With regex:
```yaml
---
- name: jedi_regex
  become: false
  gather_facts: fasle
  remote_user: k0ste
  no_log: false
  strategy: free
  hosts: localhost
  tasks:
  - name: "Step 1"
    find:
      paths: "/var/log"
      recurse: "no"
      file_type: "file"
      patterns: "^(.*)\\.log$"
      excludes: "^(pacman|lxdm)\\.log$"
      use_regex: "true"
    register: "find_result"
  - name: "Step 2"
    debug:
      var: "vars['find_result']['files']"
```

Without excludes, regex:
```
[k0ste@WorkStation tmp]$ ansible-playbook jedi_regex.yml | grep path
            "path": "/var/log/minidlna.log", 
            "path": "/var/log/pacman.log", 
            "path": "/var/log/lxdm.log", 
            "path": "/var/log/Xorg.0.log", 
```
With excludes, regex:
```
[k0ste@WorkStation tmp]$ ansible-playbook jedi_regex.yml | grep path
            "path": "/var/log/minidlna.log", 
            "path": "/var/log/Xorg.0.log", 

```

### CODE REVIEW

**Comment by k0ste** [comment]
Original https://github.com/ansible/ansible-modules-core/pull/3071

**Comment by k0ste** [comment]
ready_for_review

**Comment by k0ste** [comment]
ready_for_review

One year of rebasing...

**Comment by k0ste** [comment]
ready_for_review

**Comment by mattclay** [comment]
CI failure in `find` integration tests. Here's one example of a failure for `test/integration/targets/find/tasks/main.yml:60 / [testhost] testhost: find : validate directory results that=[u'find_test0.changed is defined', u'find_test0.examined is defined', u'find_test0.files is defined', u'find_test0.matched is defined', u'find_test0.msg is defined', u'find_test0.matched == 8', u'find_test0.files | length == 8']`:

```
{
    "assertion": "find_test0.matched == 8", 
    "changed": false, 
    "evaluated_to": false, 
    "failed": true
}
```

File: lib/ansible/modules/files/find.py [hunk 1]
```diff
@@ -45,6 +47,14 @@
             - The patterns restrict the list of files to be returned to those whose basenames match at
               least one of the patterns specified. Multiple patterns can be specified using a list.
         aliases: ['pattern']
+    excludes:
+        default: '*'
```
Comment (bcoca) [inline lib/ansible/modules/files/find.py]: default should be 'null' or 'no exclusion'

**Comment by calfonso** [comment]
@k0ste could you fix the the ci failures and rebase (again) this. Then we'll review. Thanks!

File: lib/ansible/modules/files/find.py [hunk 2]
```diff
@@ -272,6 +308,7 @@ def main():
         argument_spec=dict(
             paths=dict(type='list', required=True, aliases=['name', 'path']),
             patterns=dict(type='list', default=['*'], aliases=['pattern']),
+            excludes=dict(type='list', default=['*'], aliases=['exclude']),
```
Comment (bcoca) [inline lib/ansible/modules/files/find.py]: remove default

**Review by bcoca — DISMISSED** [review]

**Comment by k0ste** [comment]
> @k0ste could you fix the the ci failures and rebase (again) this. Then we'll review. Thanks!

Sure. Done.

**Comment by k0ste** [comment]
@calfonso ?

**Pull request merged**